vpc/routetable.tf

```diff
@@ -2,16 +2,16 @@
 # Private route table resources
 ###
 resource "aws_route_table" "private" {
-  count  = local.max_subnet_length
+  count  = local.max_subnet_length > 0 ? local.nat_gateway_count : 0
```

Comment:
Same idea of just using `local.nat_gateway_count` here if possible, and expect it to be 0 if `var.enable_eip = false`

Reply:
didn't work as expected, so i reverted back to the original method.